### DISCUSSION TITLE
Hope to add custom component documentation

### DISCUSSION THREAD

**Discussion by InfernalAzazel** [zauberzeug/nicegui#4140] [Ideas / Feature Requests]
nicegui is very amazing, I have some suggestions:

1. I hope to separate the implementation of Vue css interaction into a python library
2. When developing web, most people will choosereact because there are too many components.（We should encourage the community to create components so that we have a variety of styles of components to choose from）

- I hope that in the future we can abandon Vue react and simply use python to develop web projects to achieve projects as exciting as Vue react.

**Comment by falkoschindler** [comment]
Hi @InfernalAzazel,

Thanks for the suggestions, even though - to be honest - I don't fully understand what you're saying.

1. "Vue css interaction" doesn't really make sense as a term. Vue has its own way of handling CSS (like scoped styles and CSS modules), but it's not something you can really separate out as a standalone thing. But maybe you're trying to refer to something else entirely.
2. "Vue react" sounds like a mashup of "Vue" and "React", but these are two separate, competing frontend frameworks. NiceGUI uses Vue, not React.

**Reply by InfernalAzazel** [reply]
Sorry this is a translation error

**Reply by InfernalAzazel** [reply]
1. What I mean is that nicegui is split, and one is to implement html rendering and basic support.
2. Users can customize through the core library similar to nicegui, such as axxgui e.g bxxgui

**Reply by InfernalAzazel** [reply]
And provide documentation support for the core library, so that society can develop a variety of GUI libraries

**Reply by InfernalAzazel** [reply]
nicegui is based on https://github.com/zauberzeug/nicegui/blob/main/nicegui/templates/index.html
Unpack this and ui

**Reply by InfernalAzazel** [reply]
I'm not sure if my description is clear, you can continue to ask me, thank you very much

**Comment by zeronn2026** [comment]


**Reply by evnchn** [reply]
@zeronn2026 Is your GitHub hacked?
